File: Domain/Entities/Request.cs
using Domain.Enumerations;
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain.Entities
{
    public class Request : Entity<Request>
    {
        public string ProductId { get; set; }

        public uint NumberOfReviews { get; set; }

        public RequestStatusEnum Status { get; set; } = RequestStatusEnum.Failure;

        public DateTime Date { get; set; }

        protected override int GetHashCodeCore()
        {
            return Id.GetHashCode() ^ ProductId.GetHashCode();
        }
    }
}

File: WebUI/ClientApp/src/app/Request.ts
import { RequestStatusEnum } from './RequestStatusEnum';

class BaseRequest {
  id: number;
  numberOfReviews: number;
  status: RequestStatusEnum;
  date: Date;

  constructor(numberOfReviews: number, date: Date, status: RequestStatusEnum = RequestStatusEnum.Ongoing) {
    this.numberOfReviews = numberOfReviews;
    this.date = date;
    this.status = status;
  } 

}

export class Request extends BaseRequest{
  productId: string;

  constructor(productId: string, numberOfReviews: number, date: Date) {
    super(numberOfReviews, date);
    this.productId = productId;
  } 
}

export class BulkRequest extends BaseRequest{
  productIds: string[];
}
